Fix #6695: Move minimum deployment target to iOS 15

kylehickinson opened the pull request:
- HTTPSE is also dropped as we no longer use it on iOS 15 and higher
- Usage of `swift-markdown` has been replaced with native markdown parsing
- Added `PresentationMode` usage where manual handling of dismissing SwiftUI views were used
- Replaced custom url routing mechanism in wallet with standard `OpenURLAction`
- Replaced deprecated UTI strings with `UTType` variants
- `WKUserScript.create` removed as its no longer needed
- Many missed iOS 13 and 14 availability checks removed

## Summary of Changes

This pull request fixes #6695 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:
Manual passes on iOS 15

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

StephenHeaps reviewed (approved):
++ wallet. Just one change to remove added dismissal in `NewSiteConnectionView`

kylehickinson committed:
Fix #6695: Move minimum deployment target to iOS 15

- HTTPSE is also dropped as we no longer use it on iOS 15 and higher
- Usage of `swift-markdown` has been replaced with native markdown parsing
- Added `PresentationMode` usage where manual handling of dismissing SwiftUI views were used
- Replaced custom url routing mechanism in wallet with standard `OpenURLAction`
- Replaced deprecated UTI strings with `UTType` variants
- `WKUserScript.create` removed as its no longer needed
- Many missed iOS 13 and 14 availability checks removed

iccub reviewed (commented):
You can update README as well
